client/session.go

```diff
@@ -2213,14 +2245,35 @@ func (s *session) streamBlocksMetadataFromPeerV2(
 
 			clonedID := idPool.BinaryID(data)
 
+			var (
+				tags         ident.Tags
+				tagDecodeErr error
+			)
+			if encodedTags := elem.EncodedTags; len(encodedTags) != 0 {
+				tagDecoderCheckedBytes.Reset(encodedTags)
+				tagDecoder.Reset(tagDecoderCheckedBytes)
+				tags = make(ident.Tags, 0, tagDecoder.Remaining())
+				for tagDecoder.Next() {
+					curr := tagDecoder.Current()
+					tags = append(tags, idPool.CloneTag(curr))
```

Comment:
kinda sucks we have to allocate all the tags per peer. Ah well

Reply:
Yeah, I can do a TODO here to pool these perhaps.  We would need a wrapper type that has a ref to the pool to be able to pool the slice itself.  Although if they go into a bootstrap result, you have to hold onto them.  If you flush them out to disk then you finalize at that time, but yeah there's no way to currently track these at this time.